addons/telestation/fn_customFindSafePos.sqf

```diff
@@ -0,0 +1,161 @@
+    _defaultPos = getArray (configFile >> "CfgWorlds" >> worldName >> "Armory" >> ["positionStart", "positionStartWater"] select _this);
+    if !(_defaultPos isEqualTo []) exitWith {_defaultPos};
+		
+    _defaultPos = getArray (configFile >> "CfgWorlds" >> worldName >> "centerPosition");
+    if !(_defaultPos isEqualTo []) exitWith {_defaultPos};
+		
+    _defaultCenterPos
+};
+
+if (_checkPos isEqualTo []) then
+{
+    _checkPos = getArray (configFile >> "CfgWorlds" >> worldName >> "safePositionAnchor");
+    if (_checkPos isEqualTo []) then {_checkPos = _defaultCenterPos};
+};
+
+if (_maxDistance < 0) then 
+{
+    _maxDistance = getNumber (configFile >> "CfgWorlds" >> worldName >> "safePositionRadius");
+    if (_maxDistance <= 0) then {_maxDistance = _defaultMaxDistance};
+};
+
+private _checkProximity = _objectProximity > 0;
+private _checkBlacklist = !(_posBlacklist isEqualTo []);
+private _checkUnitBlacklist = !(_unitVisibleBlacklist isEqualTo []);
+
+_shoreMode = _shoreMode != 0;
+
+if (_checkBlacklist) then
+{
+    _posBlacklist = _posBlacklist apply 
+    {
+        // top-left, bottom-right coordinates 
+        if (_x isEqualTypeParams [[],[]]) then 
+        {
+            _x select 0 params [["_x0", 0], ["_y0", 0]];
+            _x select 1 params [["_x1", 0], ["_y1", 0]];
+            private _a = (_x1 - _x0) / 2;
+            private _b = (_y0 - _y1) / 2;
+            [[_x0 + _a, _y0 - _b], abs _a, abs _b, 0, true]
+        }
+        else
+        {
+            // other area compatible formats
+            _x call BIS_fnc_getArea
+        };
+    };
+};
+
+private _off = (_minDistance / _maxDistance) ^ 2;
+private _rem = 1 - _off;
+private _gradientRadius = 1 max _objectProximity * 0.1;
+
+for "_i" from 1 to MAX_TRIES do
+{
+    _checkPos getPos [_maxDistance * sqrt (_off + random _rem), random 360] call
+    {
+        // position is roughly suitable
+        if (_this isFlatEmpty [-1, -1, _maxGradient, _gradientRadius, _waterMode, _shoreMode] isEqualTo []) exitWith {};
+		
+        // away from other objects
+        if (_checkProximity && {!(nearestTerrainObjects [_this, [], _objectProximity, false, true] isEqualTo [])}) exitWith {};	
+		
+        // not inside something
+        if !(lineIntersectsSurfaces [AGLtoASL _this, AGLtoASL _this vectorAdd [0, 0, 50], objNull, objNull, false, 1, "GEOM", "NONE"] isEqualTo []) exitWith {};
+		
+        // not in blacklist
+        if (_checkBlacklist && {
+            {
+                if (_this inArea _x) exitWith {true}; false
+            } forEach _posBlacklist}) exitWith {};
+
+
+        if (_checkUnitBlacklist && {
+            {
+                private _pos = (AGLtoASL _this) vectorAdd [0, 0, 1.8]; // eye level to eye level check
+                private _relDir = _x getRelDir _pos;
+                if ((_relDir < 60 or _relDir > 300) && {[_x, "GEOM"] checkVisibility [eyePos _x, _pos] > 0.01}) exitWith {true}; false // check unit is facing position and position is visible
```

Comment:
Maybe introduce preprocessor constants for some constants?

Reply:
Not sure what it would solve here?

Reply:
I'd agree with @Seb105 , these are internal values (degrees of rotation) and not something user-configurable.

Reply:
Alright, if not meant to be configured.